File: MTWDM iOS Xamarin/App6/App6/ViewController.cs
using System;
using Foundation;
using UIKit;

namespace App6
{
    public partial class ViewController : UIViewController
    {
        NSDateFormatter dateFormatter = new NSDateFormatter();

        protected ViewController(IntPtr handle) : base(handle)
        {
            // Note: this .ctor should not contain any initialization logic.
        }

        partial void onRefrescarFecha(UIKit.UIButton sender)
        {
            modificaFechaHora();
        }

        public void modificaFechaHora()
        {
            dateFormatter.DateStyle = NSDateFormatterStyle.Medium;
            dateFormatter.TimeStyle = NSDateFormatterStyle.Medium;

            lblFecha.Text = dateFormatter.ToString(new NSDate());
       }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();
            // Perform any additional setup after loading the view, typically from a nib.
            modificaFechaHora();
        }

        public override void DidReceiveMemoryWarning()
        {
            base.DidReceiveMemoryWarning();
            // Release any cached data, images, etc that aren't in use.
        }
    }
}

File: MTWDM iOS Swift /app6/app6/ViewController.swift
//
//  ViewController.swift
//  app6
//
//  Created by Felipe Hernandez on 10/6/18.
//  Copyright © 2018 Felipe Hernandez. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        modificaFechaHora()
    }
    
    // MARK: - Propiedades
    let dateFormatter = DateFormatter()
    
    // MARK: - Outlets
    @IBOutlet weak var lblFecha: UILabel!
    
    // MARK: - Eventos
    @IBAction func onRefrescarFecha(_ sender: Any) {
        modificaFechaHora()
    }
    
    // MARK: - Funciones
    func modificaFechaHora() {
        dateFormatter.dateStyle = DateFormatter.Style.medium //.medium
        dateFormatter.timeStyle = .medium
        
        lblFecha.text = dateFormatter.string(from: Date())
        lblFecha.sizeToFit()
    }
}

